tt_metal/impl/device/device.cpp

```diff
@@ -321,10 +321,17 @@ void Device::initialize_device_kernel_defines()
     const metal_SocDescriptor& soc_d = tt::Cluster::instance().get_soc_desc(this->id());
     auto pcie_cores = soc_d.get_pcie_cores();
     auto grid_size = this->grid_size();
-    this->device_kernel_defines_.emplace("PCIE_NOC_X", std::to_string(pcie_cores[0].x));
-    this->device_kernel_defines_.emplace("PCIE_NOC_Y", std::to_string(pcie_cores[0].y));
-    this->device_kernel_defines_.emplace("PCIE_NOC1_X", std::to_string(tt::tt_metal::hal.noc_coordinate(NOC::NOC_1, grid_size.x, pcie_cores[0].x)));
-    this->device_kernel_defines_.emplace("PCIE_NOC1_Y", std::to_string(tt::tt_metal::hal.noc_coordinate(NOC::NOC_1, grid_size.x, pcie_cores[0].y)));
+    if (std::getenv("TT_METAL_SIMULATOR_EN")) {
```

Comment:
Passing in PCIe core to device temporary until we enable virtual coordinates but for now maybe we should:
```suggestion
    CoreCoord pcie_core = pcie_cores.empty() ? grid_size : pcie_cores[0]
    then use this as PCIE_NOC_X and PCIE_NOC_Y and calculate PCIE_NOC1*
```

Reply:
I've removed the if cond with the env var, but `pcie_noc1*` are still hardcoded to 14 or 11 like before if `pcie_cores.empty()` 

Not sure if there's a better way of doing this?